WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReaders/CardReaderSettingsViewModel.swift

```diff
@@ -24,36 +19,82 @@ enum CardReaderSettingsViewActiveAlert {
     case updating
 }
 
+struct CardReaderViewModel {
+    var displayName: String
+    var batteryStatus: String
+
+    init(_ reader: CardReader? = nil) {
+        let unknownName = NSLocalizedString(
+            "Unknown Name",
+            comment: "Displayed in the unlikely event a card reader has an empty serial number"
+        )
+        let unknownBatteryStatus = NSLocalizedString(
+            "Unknown Battery Level",
+            comment: "Displayed in the unlikely event a card reader has an indeterminate battery status"
+        )
+
+        guard reader != nil else {
+            self.displayName = unknownName
+            self.batteryStatus = unknownBatteryStatus
+            return
+        }
+
+        self.displayName = reader?.serial ?? unknownName
+
+        guard let batteryLevel = reader?.batteryLevel else {
+            self.batteryStatus = unknownBatteryStatus
+            return
+        }
+
+        let batteryLevelPercent = Int(100 * batteryLevel)
+        let batteryLevelString = NumberFormatter.localizedString(from: batteryLevelPercent as NSNumber, number: .decimal)
+        let batteryLabelFormat = NSLocalizedString(
+            "%1$@%% Battery",
+            comment: "Card reader battery level as an integer percentage"
+        )
+
+        self.batteryStatus = String.localizedStringWithFormat(batteryLabelFormat, batteryLevelString)
+    }
+}
+
 final class CardReaderSettingsViewModel: ObservableObject {
     /// Followed by CardReaderSettingsViewController to select child views and show/hide alerts.
     @Published var activeView: CardReaderSettingsViewActiveView
     @Published var activeAlert: CardReaderSettingsViewActiveAlert
-    @Published var connectedReader: CardReader?
+    @Published var connectedReaderViewModel: CardReaderViewModel
+    @Published var foundReadersViewModels: [CardReaderViewModel]
 
-    var knownReaders: [CardReader]
-    var foundReader: CardReader?
-
-    var timer: Timer? // TODO Remove
-    var batteryTimer: Timer? // TODO Remove
+    private var connectedReader: CardReader? {
+        didSet {
+            updateConnectedReaderViewModel()
```

Comment:
Update the connectedReader, get a published update to the VM. Same thing below for foundReaders